app/models/job_proxy_dispatcher.rb

```diff
@@ -197,11 +263,29 @@ def busy_proxies
     end
   end
 
-  def active_scans
-    @active_scans ||= begin
-      actives = Hash.new(0)
-      actives[@zone] = Job.where(:zone => @zone, :dispatch_status => "active").where.not(:state => "finished").count
-      actives
+  def active_scans_by_zone(target_class, count = true)
+    class_name = target_class.base_class.name
+    actives = Hash.new(0)
+     jobs = Job.where(:zone => @zone, :dispatch_status => "active", :target_class => class_name)
+            .where.not(:state => "finished")
+    actives[@zone] = count ? jobs.count : jobs
+    actives
+  end
+
+  def active_vm_scans_by_zone
+    @active_vm_scans_by_zone ||= active_scans_by_zone(VmOrTemplate)
+  end
+
+  def active_container_scans_by_zone_and_ems(images_to_ems)
+    @active_container_scans_by_zone_and_ems ||= begin
+      memo = Hash.new do |h, k|
+        h[k] = Hash.new(0)
+      end
+
+      active_scans_by_zone(ContainerImage, false)[@zone].each do |job|
+        memo[@zone][images_to_ems[job.target_id]] += 1
+      end
+      memo
```

Comment:
Please add tests for these methods.  It should be very easy to setup the db to test that mixtures of container image/vm or template jobs are detected in the correct "active*scans" method.